relay-general/src/processor/selector.rs

```diff
@@ -294,10 +298,7 @@ fn handle_selector(pair: Pair<Rule>) -> Result<SelectorSpec, InvalidSelectorErro
         Rule::NotSelector => Ok(SelectorSpec::Not(Box::new(handle_selector(
             pair.into_inner().next().unwrap(),
         )?))),
-        rule => Err(InvalidSelectorError::UnexpectedToken(
-            format!("{rule:?}"),
-            "a selector",
-        )),
+        _ => Ok(SelectorSpec::Other(pair.as_str().to_string())),
```

Comment:
so far unable to actually find a case where this code gets executed with unit tests, using the from_str method. 

it either gets parsed into a `SelectorSpec::Path`, or it returns `InvalidSelectorError::ParseError`

Reply:
If it can return an `InvalidSeelctorError`, it seems this branch would catch it.

Reply:
I'm not sure what you mean here exactly 

About this `InvalidSelectError`, among the variants:

```
pub enum InvalidSelectorError {
    #[error("invalid selector: deep wildcard used more than once")]
    InvalidDeepWildcard,

    #[error("invalid selector: wildcard must be part of a path")]
    InvalidWildcard,

    #[error("invalid selector: {0}")]
    ParseError(Box<Error<Rule>>),

    #[error("invalid selector: invalid index")]
    InvalidIndex,

    #[error("invalid selector: unknown value")]
    UnknownType,

    #[error("parser bug: consumed {0} (expected {1})")]
    UnexpectedToken(String, &'static str),
}
```

the ones that to me seems relevant for forward compatiblity are: ParseError, UnknownType, and UnexpextedToken, if this is right we could remove them and instead put them all into Other(), or maybe even somehow also pass along the errors themselves?